src/components/views/elements/MemberEventListSummary.js

```diff
@@ -144,87 +144,105 @@ module.exports = React.createClass({
         );
     },
 
-    render: function() {
-        let summary = null;
+    shouldComponentUpdate: function(nextProps, nextState) {
+        return (
+            nextProps.events.length !== this.props.events.length ||
```

Comment:
hrm. this is going to mean that, when the summary is expanded, any updates to the eventTiles in `children` won't be rerendered. I don't know if that's a problem in practice, but it certainly sounds like a potential future problem.

Maybe always return true if `nextState.exanded` or `this.state.expanded` is true, or if there are few events?

I might be missing why a `shouldComponentUpdate` is actually necessary here. Having removed the O(N^2) stuff in render I think it might be more hassle than it is worth. Or am I missing something?